src/Uno.UI.Composition/Composition/CompositionSurfaceBrush.skia.cs

```diff
@@ -81,6 +81,11 @@ internal override void UpdatePaint(SKPaint fillPaint, SKRect bounds)
 			{
 				var backgroundArea = GetArrangedImageRect(new Size(scs.Image!.Width, scs.Image.Height), bounds);
 
+				if (backgroundArea.Width <= 0 || backgroundArea.Height <= 0)
+				{
+					return;
+				}
```

Comment:
Instead of this, I'd do it this way:

```csharp
var resizedBitmap = scs.Image.ToSKBimap();
if (backgroundArea.Width > 0 && backgroundArea.Height > 0)
{
    resizedBitmap = resizedBitmap.Resize(...);
}

var resizedImage = SKImage.FromBitmap(resizedBitmap);
// ...
```

The reason I prefer this way is to ensure as we can that there are no surprises regarding any behavioral changes. For example, I'm not sure if it's possible this return can keep the `fillPaint` unchanged when it should have changed.

Reply:
Makes sense. It's important that the `fillPaint` is cleared before sending it to `UpdatePaint` anyway, but sure.

Reply:
Shouldn't we check for null at the location of the original issue? Was the null only returned because of the size?

Reply:
The null is returned by the `Resize` call. If we'll just ignore the resizing if `Resize` is null, that could mask an issue if it's something other than the dimensions.

Reply:
Actually now thinking about it, proceeding without resizing if `backgroundArea` is empty looks really weird to me. If the area is empty, we shouldn't be doing anything. At the very least, we shouldn't be proceeding with a full-size image. I will prefer to keep the PR as is.

Reply:
It's a behavior change that's unrelated to the `ArgumentNullException`. If the current behavior wrong, we need to have a scenario showing that (and preferably a runtime test)

Reply:
It's not really a behavior change. The invariant was and should continue to be that after whatever processing we do, the image we render is now sized at `backgroundArea`. Before #16816, we always resized but with terrible quality. After #16816, we crash if resizing failed. With this PR, we don't draw at all if we can't resize. Proceeding with painting with an image shader that doesn't match the `backgroundArea` would be the only change that breaks this invariant.

Reply:
Then maybe we should cleear any possibly existing Shader on fillPaint?

Reply:
Yes, good point.